src/transformers/modeling_utils.py

```diff
@@ -532,23 +595,37 @@ def _load_state_dict_into_meta_model(model, state_dict, loaded_state_dict_keys,
     # - Is there a situation where some keys aren't in `loaded_state_dict_keys` and in which case
     #   they won't get loaded.
 
-    if is_deepspeed_zero3_enabled():
-        raise ValueError("low_cpu_mem_usage arg cannot currently be used with DeepSpeed ZeRO-3")
-
     error_msgs = []
 
-    # materialize state_dict entries one by one on CPU
-    for k in loaded_state_dict_keys:
-        if k in state_dict:
-            submodule, param_name = find_submodule_and_param_name(model, k, start_prefix)
-            if submodule is not None:
-                param_dtype = getattr(submodule, param_name).dtype
-                new_val = state_dict[k].to(param_dtype)
-                if isinstance(getattr(submodule, param_name), torch.nn.Parameter):
-                    new_val = torch.nn.Parameter(new_val)
-                setattr(submodule, param_name, new_val)
+    for param_name, param in state_dict.items():
+        # First part of the test is always true as load_state_dict_keys always contains state_dict keys.
+        if param_name not in loaded_state_dict_keys or param_name not in expected_keys:
```

Comment:
First part of the test is left to be the same as before, but as said in the comment, it shouldn't be necessary as:
- `loaded_state_dict_keys = state_dict.keys()` when the checkpoint is one file
-  `loaded_state_dict_keys` contains `state_dict.keys()` when the checkpoint is sharded